streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/HighAvailabilityTaskAssignorTest.java

```diff
@@ -41,132 +54,107 @@
 import static org.easymock.EasyMock.replay;
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
-import java.util.UUID;
-import org.apache.kafka.streams.processor.TaskId;
-import org.apache.kafka.streams.processor.internals.assignment.AssignorConfiguration.AssignmentConfigs;
-import org.easymock.EasyMock;
-import org.junit.Test;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
 
 public class HighAvailabilityTaskAssignorTest {
```

Comment:
I made a bunch of changes to this test, because it was pretty brittle with respect to changes in the HighAvailabilityTaskAssignor.

Namely:
* by mocking ClientState instead of constructing the desired state, we'd throw NPEs in the assignor that could never happen in production if the assignor is modified in a way that happens to need to query the ClientState a little differently than the tests anticipate (for example, the lags were not always mocked). Now, we just construct a ClientState for each client, representing the desired scenario and make assertions on the resulting assignment.
* by relying heavily on shared mutable fields inserted into shared mutable collections to build a shared assignor, it was relatively hard to read the tests "cold" and figure out how they were configured. I minimized the shared fields, and eliminated the mutability, so now the tests are pretty much self-contained. This is a bit opinionated, but I feel like it's the right thing to do. It seemed like my non-trivial ramp-up period where I figured out how the tests were actually constructed would decay pretty rapidly, and if I have to come back to this test in a couple of months, I didn't want to have to start learning all over again.